articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+All Q# callables and types (both those you define and those intrinsic to the language) are defined within namespaces.
+To access them, the compiler needs to be made aware of which namespaces to open.
+
+So, the full Q# file containing our operation consists of defining our own namespace, opening the namespaces for the operations our operation uses, and then our operation:
+
+```qsharp
+namespace NamespaceName {
+    open Microsoft.Quantum.Intrinsic;     // for the H operation
+    open Microsoft.Quantum.Measurement;   // for MResetZ
+
+    operation MeasureSuperposition() : Result {
+        using (q = Qubit()) { 
+            H(q);
+            return MResetZ(q);
+        }
+    }
+}
+```
+
+### Execution of a Q# program
+
+Now the general execution model of a Q# program becomes clear.
+
+<br/>
+<img src="./images/general_execution_model.png" alt="Q# program execution diagram" width="400">
+
+Firstly, the specific operation to be executed has access to any other callables and types defined in the same namespace.
+Thanks to `open` statements, it can also access those from any of the [Q# libraries](xref:microsoft.quantum.libraries). 
+
+The operation itself is then executed on a *[target machine](xref:microsoft.quantum.machines)*.
+In the future, one such target machine will be a real quantum computer, but there are currently multiple simulators available, each with a particular use.
+For our purposes here, the most useful target machine is instance of the full-state simulator, `QuantumSimulator`, which calculates the program's behavior as if it were being executed on a noise-free quantum computer.
+
+So far, we've described what happens when a specific Q# operation is being executed.
+Regardless of whether Q# is used in a standalone application or with a host program, this general process is more or less the same---hence the QDK's flexibility.
+The differences between the methods therefore reveal themselves in *how* that Q# operation is called to be executed, and in what manner any results are returned.
+
+First, we discuss how this done with the Q# standalone application from the command line, and then proceed to using Python and C# host programs.
+We reserve the standalone application of Q# Jupyter Notebooks for last, because unlike the first three, it does not actually require a Q# file.
+
+
+## Q# from the command line
+One of the easiest ways to get started writing Q# programs is to avoid worrying about separate files and a second language altogether.
+Using Visual Studio Code or Visual Studio with the QDK extension allows for a seamless work flow in which we run Q# operations from only a single Q# file.
+
+For this, we will ultimately invoke the program's execution by entering
+```bash
+dotnet run
+```
+in the command line.
+Note that the terminal's directory location needs to be the same as the Q# file, so we recommend simply using the integrated terminal in VS Code or Visual Studio.
```

Comment:
You can also use `--project` to run something in a different folder. Also, I don't think VS2019 has an integrated terminal _per se_.